packages/hs-react-ui/src/components/Button/Button.tsx

```diff
@@ -1,66 +1,96 @@
 import React from 'react';
+import UnstyledIcon from '@mdi/react';
+import { mdiLoading } from '@mdi/js';
 import styled, { StyledComponentBase } from 'styled-components';
-import colors from '../../enums/colors';
-import fonts from '../../enums/fonts';
+import { ButtonContainers, ButtonTypes, ButtonContainer, Progress } from './ButtonContainers';
 
-const Text = styled.span``;
+export type ButtonProps = {
+  StyledContainer?: string & StyledComponentBase<any, {}, { elevation?: number; color?: string }>;
+  iconPrefix?: string | JSX.Element;
+  iconSuffix?: string | JSX.Element;
+  isLoading?: boolean;
+  isProcessing?: boolean;
+  children?: string | Node;
+  elevation?: number;
+  type?: string;
+  color?: string;
+  fillColor?: string;
+  onClick: (...args: any[]) => void;
+};
 
-export enum ButtonTypes {
-  primary = 'primary',
-  secondary = 'secondary',
-  default = 'default',
-  destructive = 'destructive',
-}
+const Icon = styled(UnstyledIcon)``;
 
-export const ButtonContainer = styled.button`
+const Text = styled.span`
   display: inline-block;
+  margin-top: -8px;
+  margin-bottom: -8px;
+`;
 
-  ${fonts.body}
-  font-size: 1em;
-  line-height: 0;
-
-  padding: 1.25em 1em;
-  border-radius: 0.25em;
-
-  outline: 0 none;
-  border: 0 none;
-
-  cursor: pointer;
-
-  background-color: ${colors.grayXlight};
-  color: ${colors.grayDark};
+const LeftIconContainer = styled(Text)`
+  margin-right: 0.25rem;
 `;
 
-export type ButtonProps = {
-  StyledContainer?: string & StyledComponentBase<any, {}>;
-  icon?: any;
-  isLoading?: any;
-  children?: string | Node;
-  onClick: (...args: any[]) => void;
-};
+const RightIconContainer = styled(Text)`
+  margin-left: 0.25rem;
+`;
 
 const Button = ({
-  StyledContainer = ButtonContainer,
-  icon,
+  StyledContainer,
+  iconPrefix,
+  iconSuffix,
   isLoading,
+  isProcessing,
   children,
+  elevation = 0,
+  type = ButtonTypes.default,
+  color,
+  fillColor,
   onClick,
 }: ButtonProps) => {
+  const Container = StyledContainer || ButtonContainers[type];
   return isLoading ? (
-    <StyledContainer>
-      <Text>
-        {isLoading.children} {isLoading.icon}
-      </Text>
-    </StyledContainer>
+    <Container data-testid="hsui-button" elevation={elevation} color={color} fillColor={fillColor}>
+      <Progress />
+    </Container>
   ) : (
-    <StyledContainer onClick={onClick}>
+    <Container
+      data-testid="hsui-button"
```

Comment:
could this be `data-test-id`?

Reply:
Yes, I'll do this. It's worth noting, however, that there is a very tiny configuration change required for the @testing-library/react: `configure({testIdAttribute: 'data-test-id'))` which may be required in other test files as well.

https://testing-library.com/docs/dom-testing-library/api-queries#overriding-data-testid